plugin/pkg/auth/authorizer/node/graph.go

```diff
@@ -393,6 +396,16 @@ func (g *Graph) AddPod(pod *corev1.Pod) {
 			g.addEdgeToDestinationIndex_locked(e)
 		}
 	}
+
+	for _, podResourceClaim := range pod.Spec.ResourceClaims {
+		claimName := resourceclaim.Name(pod, &podResourceClaim)
```

Comment:
it looks like resourceclaim.Name returns `pod.Name + claim.Name` ... since both of those can contain `-` characters, the following would conflict, right?

* pod `foo`, claim `bar-baz`
* pod `foo-bar`, claim `baz`

I also don't see validation that ensures any of the following:
* ResourceClaimName is valid (passes the same validation we apply in ValidateClaim)
* ResourceClaimTemplateName is valid (passes the same validation we apply in ValidateClaimTemplate)
* the combined `pod.Name + "-" + claim.Name` is valid (passes the same validation we apply in ValidateClaim, *especially* length requirements)

Reply:
> The conflict between foo + bar-baz and foo-bar and baz was mentioned before. As both pods come from the same namespace, the user would be responsible for ensuring that no such clash occurs. Would it a) be possible to tighten the validation and b) useful?

I don't see why we would expose the user to sharp edges like that if we can avoid it with requirements for templated / directly used claim object names

Reply:
> I don't see why we would expose the user to sharp edges like that if we can avoid it with requirements for templated / directly used claim object names

So that's a "yes" for not allowing "-" in the `pod.spec.resourceClaims[*].name` field, right? Is it okay to tighten that validation in 1.27? I think it's okay as we are making some other "action required" changes, I just wanted to be sure.

> We should validate the fields we have. If we have a reference in core, we should validate it.

/cc @thockin 

I think you were the one who argued that core shouldn't validate against requirements defined by the v1alpha1 resource API.

I can add it back if the consensus is that we should do that.

Reply:
Can we instead use a separator that is disallowed in both? i.e. store the key as:
```
pod.Name + "#" + claim.Name
```

From a user's perspective it will feel unnecessarily prohibitive to not be allowed to use dashes in the claim name.

Reply:
the issue isn't the key in the graph, it's the actual API object name

Reply:
Ah right.

Reply:
We can concatenate with `--` and then not allow that inside the claim name. The downside is that a generated `my-pod--my-claim` will look a bit odd. Or perhaps that's an upside? It further reduces the risk that someone accidentally creates a ResourceClaim with such a name and then has a conflict.

Reply:
I don't think we should lean on character-set trickery here.  I just went on a rant about this on Twitter - names are for humans.  Parsing names for metadata is a bad idea.

I'd much prefer that the name not be deterministic, and instead associate it by labels.

E.g.

Pod foo, claim bar-baz => "foo-random", + labels {"dra.k8s.io/pod: "foo", "dra.k8s.io/resource-claim: bar-baz"}
Pod foo-bar, claim baz => "foo-bar-random", + labels {"dra.k8s.io/pod: "foo-bar", "dra.k8s.io/resource-claim: baz"}

You need to handle the edge-case of getting more than one result, but that's solvable with docs.

Or instead of labels, generate a random name and store it in status?

Reply:
I was discussing such a solution earlier with Jordan (see above) and [the outcome was](https://github.com/kubernetes/kubernetes/pull/116254#issuecomment-1463202076):
> If the name can be deterministic, that would be much better both for the kubelet discovering and authorizing.

The reason why just labels doesn't work well is that a kubelet on a specific node won't know which ResourceClaim it needs to read for a pod - it'll have to list them with a label filter, which just won't be very efficient.

Storing the generated name in the status would work, but does anyone here really have the motivation for another core/v1 API change?

Reply:
> a kubelet on a specific node won't know which ResourceClaim it needs to read for a pod

OK, that's a fair point - ideally "The set of resources which need to be read in order to run a pod are directly linked by the pod".  Good principle.

> Storing the generated name in the status would work, but does anyone here really have the motivation for another core/v1 API change?

I do this all day.  v1 changes aren't SO bad given that you already have a gate in place.

@liggitt what think?

Reply:
Either deterministic mappings or explicit references from pod status work in my view.

Putting the name of the spawned claim(s) in pod status for kubelets to discover and for the node authorizer to authorize on *could* work. `NodeRestriction` admission would need an update to disallow nodes adding names of claims in pod status, but that's not hard.

I don't have a good sense for the trust level we're placing in the controller(s) handling these claim objects and whether we'd be happy letting those controller(s) write to status of arbitrary pods. I haven't followed the design enough to know if we expect a single central controller running on the control plane (which we'd be happy to let write pod status), or if this is a distributed bring-your-own-controller model (where we would *not* want to hand out pod status update permissions like candy).

Reply:
It's a bring-your-own-controller model

Reply:
sorry, I meant the controller creating the resourceclaim object and updating pod status

Reply:
Right -- that's launched by the `kube-controller-manager` and implemented here:
https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/resourceclaim/controller.go